generators/generator-base.js

```diff
@@ -1355,7 +1355,7 @@ module.exports = class extends PrivateBase {
      * @param {string} prodDatabaseType - database type
      */
     getJoinTableName(entityName, relationshipName, prodDatabaseType) {
-        const joinTableName = `${this.getTableName(entityName)}_${this.getTableName(relationshipName)}`;
+        const joinTableName = `${this.getTableName(entityName)}__${this.getTableName(relationshipName)}`;
```

Comment:
This sounds like a breaking change, isn't it?

Reply:
We can add a config field for this separator.

If its an upgrade, we set as _ otherwise '__'.
An alternative is set a prefix like 'rel_' for relationship tables.

Reply:
@murdos @MathieuAA, I reverted the dirty workaround.
If we just change the table name, there will be no upgrade path from jhipster 6 => jhipster 7.
All relationships tables, primary keys, and references must be fixed by hand.
I don't like either, but I think we really should use the workaround.

Reply:
I think I see your point @mshima. Thanks for the suggestions.
I don't know how this use case is handled in the wild as I've always seen things like `a_b` for joint tables, but I'm sure there must be an elegant solution somewhere...
I like the `rel_` as it's explicit it concerns a relationship.
 
@jdubois didn't you love to play with DBs a long time ago? Do you have an an idea for this issue?

Reply:
I like prefixing relationships with `rel_` too but adding prefix of suffix, will not fix relations like
relation_name -> other = relation_name_other
rel -> name_other = relation_name_other

Reply:
@mshima Argh! You're right. Any character as separator won't do...